Conform models to Sendable in preparation for Swift 6

hyungsangpark opened the pull request:
# Description
Swift 6 enforces stricter concurrency checking, including checks for the conformance to Sendable protocol.
Since API models are frequently used across both synchronous and asynchronous contexts, conforming the model objects with Sendable protocol will ensure we follow the Swift 6 concurrency requirements. 
In addition, this will reduce the amount of errors you'll see when you enable "Strict Concurrency Checking" for Swift 6 preparation.

In addition to the added conformance to the Sendable protocol, this PR also adds two additional reserved keywords: `Sendable` and `Hashable`. They are both used in codebase to indicate conformance to `Swift.Sendable` and `Swift.Hashable`. 

One thing to note is that in the templates, they are actually referenced using absolute references, meaning even if the API yaml were to use the keywords `Sendable` or `Hashable`, the generator would have no problem producing a non-failing code. I did, however, add it as a reserved keyword for consistency, as similar protocols such as `Codable`, `Decodable` and `Encodable` were in the reserved keyword already.

hyungsangpark committed:
Conform models to Sendable in preparation for Swift 6

karlvr commented on src/index.ts:
```diff
@@ -84,7 +84,7 @@ const RESERVED_WORDS = [
 	'internal', 'is', 'lazy', 'left', 'let', 'mutating', 'nil', 'none', 'nonmutating', 'open', 'operator', 'optional', 'override', 'postfix', 'precedence', 'prefix', 'private',
 	'protocol', 'public', 'repeat', 'required', 'rethrows', 'return', 'right', 'self', 'set', 'static', 'struct', 'subscript', 'super', 'switch', 'throw', 'throws', 'true',
-	'try', 'typealias', 'unowned', 'var', 'weak', 'where', 'while', 'willSet',
+	'try', 'typealias', 'unowned', 'var', 'weak', 'where', 'while', 'willSet', 'Sendable', 'Hashable',
```
This is maybe a slightly odd place to add these... how about we add these in a new line underneath the `LocalDate` family? The line you've added it to finishes the language keywords chunk of reserved words I reckon.

BUT... I reckon we don't add these as reserved words if we don't need to. If you reckon we don't need to add others in here then how about a separate PR to remove those?

hyungsangpark replied:
Maybe they aren't necessary, but I could see that we also have `Encodable` and `Decodable` listed above in the list, so I figured it'll be better and probably safer to have it in, of course in a new line below `LocalDate` family.

Resolved in 07dfa67

hyungsangpark committed:
fix: move down Sendable and Hashable keywords a line

they aren’t necessarily belong in Swift’s reserved keywords category; they are more like additional symbols that we are using

hyungsangpark committed:
Conform requestable and results as Sendable

hyungsangpark committed:
Mark certain models in support as Sendable

hyungsangpark committed:
Convert Api to struct

to conform Sendable

hyungsangpark committed:
fix: use absolute reference to avoid namespacing issue

hyungsangpark committed:
gitignore: added test and system files

hyungsangpark committed:
fix: remove open-ness from functions

hyungsangpark committed:
fix: update commenting convention

karlvr commented on .changeset/neat-jobs-bow.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@openapi-generator-plus/swift-client-generator": patch
```
This seems like maybe it's more of a minor change than a patch? These messages end up in the CHANGELOG so maybe we can flesh out the message _a little_... like did we change Api models to structs from classes?

hyungsangpark replied:
How about "Convert Api calling classes to structs"?

hyungsangpark replied:
I've changed Api classes, e.g. FilmsApi, ProductionsApi, MembersApi etc to structs so that they can conform to Sendable.

oh if so, could "Convert Api calling classes to structs and conform to Sendable" work as well?

hyungsangpark replied:
Resolved in [5835203](https://github.com/karlvr/openapi-generator-plus-swift-client-generator/pull/12/commits/5835203725f7b16a21dd0554e1d2cc3d8f414c35)

karlvr commented on .changeset/pretty-pumpkins-approve.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@openapi-generator-plus/swift-client-generator": patch
```
This also seems like at least a minor feature. https://semver.org is a good reference for how to think about major, minor and patch-level changes... just the summary section at the top of the page.

Could you also take a look at the changelog message to see if you can make it more comprehensible? Checkout the `CHANGELOG.md` file as an example of what we have in there already.

hyungsangpark replied:
```
Added Sendable conformance to requestable and result objects in Api classes
```

How does this sound?

hyungsangpark replied:
Resolved in [5835203](https://github.com/karlvr/openapi-generator-plus-swift-client-generator/pull/12/commits/5835203725f7b16a21dd0554e1d2cc3d8f414c35)

karlvr commented on .changeset/seven-ties-tickle.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@openapi-generator-plus/swift-client-generator": patch
```
Probably also at least minor. Maybe review message as well with others.

hyungsangpark replied:
```
Conform enum and data models to Sendable in preparation for Swift 6
```

How does this sound?

hyungsangpark replied:
Resolved in [5835203](https://github.com/karlvr/openapi-generator-plus-swift-client-generator/pull/12/commits/5835203725f7b16a21dd0554e1d2cc3d8f414c35)

karlvr commented on .changeset/sharp-grapes-repeat.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@openapi-generator-plus/swift-client-generator": patch
```
Maybe minor, and maybe tweak message. Perhaps it's not too much to be specific about the models we made conform to `Sendable`. Because these messages are going to all end up on separate lines in the changelog then we should think about how they read in there. Maybe we don't need four of them if this is one PR... maybe you can condense down to one.

hyungsangpark replied:
Hmm in the PR I've changed these files:

```
/security/OAuthAccessToken.swift
/support/Configuration.swift
/support/NameValuePair.swift
/support/OAuthConfiguration.swift
/support/RetryConfiguration.swift
```

So I suppose I could name them, or perhaps tidy it up so that it's worded like this?
```
Mark api security and support models as Sendable
```

hyungsangpark replied:
Resolved in [5835203](https://github.com/karlvr/openapi-generator-plus-swift-client-generator/pull/12/commits/5835203725f7b16a21dd0554e1d2cc3d8f414c35)

karlvr commented on src/index.ts:
```diff
@@ -86,6 +86,7 @@ const RESERVED_WORDS = [
 	'protocol', 'public', 'repeat', 'required', 'rethrows', 'return', 'right', 'self', 'set', 'static', 'struct', 'subscript', 'super', 'switch', 'throw', 'throws', 'true',
 	'try', 'typealias', 'unowned', 'var', 'weak', 'where', 'while', 'willSet',
 	'LocalDate', 'LocalTime', 'OffsetDateTime', 'Decimal', 'String', 'Void', 'File', 'FormData',
+	'Sendable', 'Hashable',
```
I don't think we need to add this. The names we've added here are either reserved words or the names of types that _we_ generate in our module. We must prevent generated files using those names as we cannot reference them unambiguously within the module. Because we use `Swift.Sendable` etc, we don't have the same problem with referencing modules _outside_ our module.

I don't believe we need `Decimal`, `String`, or `Void` in this list because they are also external to this module. But we don't need to fix that here.

hyungsangpark replied:
Huh, right, so I suppose LocalDate, LocalTime, OffsetDateTime could also be accessed with an absolute reference and potentially be removed from here. Obviously in a different PR.

I can remove Sendable and Hashable then.

hyungsangpark replied:
Resolved in [827b022](https://github.com/karlvr/openapi-generator-plus-swift-client-generator/pull/12/commits/827b022ba59b3f18e9139bfe2e2649397f401b55)

hyungsangpark committed:
revert: remove incorrectly added reserved keywords

Sendable and Hashable

hyungsangpark committed:
fix: update changeset level and description
